app/services/invoices/calculate_fees_service.rb

```diff
@@ -293,7 +332,29 @@ def should_create_charge_fees?(subscription)
       true
     end
 
+    def should_create_fixed_charge_fees?(subscription, boundaries)
+      fee_count = subscription.fees
+        .fixed_charge
+        .includes(:invoice)
```

Comment:
we don't need to eager load `invoices` as the query is only checking the `invoice_id` that is already fees table column.